AndroidSDKLocation/src/main/AndroidManifest.xml

```diff
@@ -2,6 +2,10 @@
           xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-sdk android:minSdkVersion="14"/>
+    <!-- Permissions for geofencing. -->
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
```

Comment:
It seems like you only need ACCESS_FINE_LOCATION  or ACCESS_COARSE_LOCATION  by this https://developer.android.com/guide/topics/location/strategies.html#Permission

Reply:
there were some cases when you need both, I prefer to keep both.